router.go

```diff
@@ -334,21 +333,45 @@ func (r *Router) Find(method, path string, c Context) {
 	cn := r.tree // Current node as root
 
 	var (
-		search  = path
-		child   *node         // Child node
-		n       int           // Param counter
-		nk      kind          // Next kind
-		nn      *node         // Next node
-		ns      string        // Next search
-		pvalues = ctx.pvalues // Use the internal slice so the interface can keep the illusion of a dynamic slice
+		search      = path
+		searchIndex = 0
+		n           int           // Param counter
+		pvalues     = ctx.pvalues // Use the internal slice so the interface can keep the illusion of a dynamic slice
 	)
 
-	// Search order static > param > any
-	for {
-		if search == "" {
-			break
+	// backtracking happens when we reach dead end when matching nodes in the router tree. To backtrack we set
```

Comment:
Suggestion for comment. Not sure I got it right, take what makes sense to you:
```
  // Backtracking is needed when a dead end (leaf node) is reached in the router tree.
	// To backtrack the current node will be changed to the parent node and the next kind for the
	// router logic will be returned based on fromKind or kind of the dead end node (static > param > any).
	// For example if there is no static node match we should check parent next sibling by kind (param). 
	// Backtracking itself does not check if there is a next sibling, this is done by the router logic.
```

Reply:
changed